### DISCUSSION TITLE
Proposal: Switch to Ubuntu for .NET convenience tags

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/dotnet-docker#5709] [Announcements]
We use Debian images for `8.0` and similar convenience tags. Debian has served us very well, except that Debian and .NET release cycles align poorly. We should switch the convenience tags to Ubuntu to remedy this problem. We will continue to publish Debian images. This proposal is for .NET 10.

Proposed pattern:

- `10.0` tags will be based on Ubuntu 24.04
- `11.0` tags will be based on Ubuntu 26.04
- `12.0` tags will be based on Ubuntu 26.04

After this change, we'd expect to have the following tags (for .NET 10) at .NET 10 GA.

- `10.0`: Ubuntu 24.04
- `10.0-noble`: Ubuntu 24.04
- `10.0-trixie`: Debian 13
- `10.0-alpine`: Alpine 3.x

Each of those tag families has other variants, for version and architecture.

After Debian 14 is released, we'd expect the following:

- `10.0`: Ubuntu 24.04
- `10.0-noble`: Ubuntu 24.04
- `10.0-trixie`: Debian 13
- `10.0-debian14`: Debian 14
- `10.0-alpine`: Alpine 3.x

After Debian 13 goes EOL, we'd expect the following:

- `10.0`: Ubuntu 24.04
- `10.0-noble`: Ubuntu 24.04
- `10.0-debian14`: Debian 14
- `10.0-alpine`: Alpine 3.x

Note: `debian14` is a placeholder for the Debian 14 codename.

## Details

The convenience tags are important because they get a lot of usage precicely because they are covenient to use. As a result, they should offer a "no regrets" experience. There can always be better or more optimized choices, but there should be nothing about these tags that could be considered a poor choice.

There are two challenges with Debian images for us, all related to release cycle alignments:

- The new Debian LTS release is ready just a couple/few months ahead of the new .NET LTS release.
- Debian LTS releases have three years of full mainline support, matching the three year .NET LTS lifecycle, but Debian always ships first, creating a mainline support gap.

The first point is acceptable, but undesirable. We would prefer to adopt an OS that was release >= 6 months earlier than a .NET release. The second point is a significant problem for convenience tags. We've always seen Debian as the Docker Hub default and stuck with it for that reason. In retrospect, we should have resolved this a long time ago.

In contrast, with Ubuntu:

- The newest Ubuntu LTS is ready just over one and a half years ahead of a .NET LTS.
- Ubuntu LTS releases have five years of standard support, which would outlast a .NET LTS release.

Note: Some people consider the [`.1` release](https://lists.ubuntu.com/archives/ubuntu-announce/2022-August/000282.html) to be where LTS really starts for Ubuntu. That comes out later, making the gap closer to a year with a significant increase in quality.

We can test how this would have worked out for .NET 8 if we'd used this model with Ubuntu 22.04.

- Ubuntu 22.04 shipped in April, 2022.
- .NET 8 shipped in November, 2023.
- .NET 8 goes EOL in November, 2026.
- Ubuntu 22.04 exits standard support on June 2027.
- Ubuntu 22.04 goes EOL in April 2034.

That would have worked well. That is, however, not what we did. It is what we propose to do for .NET 10 and Ubuntu 24.04.

Some folks conclude that current practice is better since they would prefer a newer OS. However, it should be noted that we observe that Ubuntu releases (at at least in some cases) are quicker to adopt eocosystem changes. For example, Ubuntu (22.04) adopted OpenSSL 3 before Debian (12).

Lifecycle information:

- https://github.com/dotnet/core/blob/main/releases.md
- https://www.debian.org/releases/
- https://wiki.ubuntu.com/Releases

Related issues:

- https://github.com/dotnet/dotnet-docker/discussions/5710
- https://github.com/dotnet/dotnet-docker/issues/647
- https://github.com/dotnet/dotnet-docker/issues/2576
- https://github.com/dotnet/dotnet-docker/issues/4322
- https://devblogs.microsoft.com/dotnet/announcing-net-6/#debian-11-bullseye
- https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-1/#net-container-images

**Comment by lbussell** [comment]
> Proposed pattern:
>
> - `10.0` tags will be based on Ubuntu 24.04
> - `11.0` tags will be based on Ubuntu 24.04
> - `12.0` tags will be based on Ubuntu 26.04

.NET 10 Ships in 11-2025. If .NET 11 ships in 11-2026, why wouldn't it have a default of Ubuntu 26.04? @richlander

**Reply by richlander** [reply]
That's a great point! I think that was an off-by-one error. Updated the text.
